thanosql/resources/_query.py

```diff
@@ -147,6 +149,126 @@ def execute(
+            In this case, values should be a list of tuples. For example, we have
+            the following values:
+            ```
+            [(1, "gangnam", "seoul"), (2, "haeundae", "busan")]
+            ```
+            and we leave template empty. The original query looks like
+            ```
+            VALUES
+                {{ val }}
+            ```
+            The rendered query will look like
+            ```
+            VALUES
+                (1, 'gangnam', 'seoul'),
+                (2, 'haeundae', 'busan')
+            ```
+            Meanwhile, if we have the following named values:
+            ```
+            [
+                {"id": 1, "name": "gangnam", "city": "seoul"},
+                {"id": 2, "name": "haeundae", "city": "busan"},
+            ]
+            ```
+            and we have the following template:
+            ```
+            ({{ id }}, CONCAT(INITCAP({{ name }}), '-gu'), INITCAP({{ city }}))
+            ```
+            the rendered query will be:
+            ```
+            VALUES
+                (1, CONCAT(INITCAP('gangnam', '-gu')), INITCAP('seoul')),
+                (2, CONCAT(INITCAP('haeundae', '-gu')), INITCAP('busan'))
+            ```
+        page_size: int, default 100
+            The maximum number of rows (set of values) to be sent in one statement.
+            If not specified, it defaults to 100.
+        schema: str, optional
+            The schema of the table to save the query results in. If not specified
+            and table_name is also empty, it defaults to "qm". If table_name is
+            specified, it defaults to "public".
+        table_name: str, optional
+            The name of the table to save the query results in. If not specified
+            while the query produces a result, an automatic table name will be given.
+        overwrite: bool, optional
+            Whether to overwrite the table if a table with the same `table_name`
+            and `schema` already exists. If not specified, the value is False.
+        max_results: int, default 100
+            The maximum number of records to be returned by the response QueryLog.
+            If not specified, it defaults to 100.
+
+        Returns
+        -------
+        QueryLog
+            A query log object containing details about the results of the
+            executed query.
+
+        Raises
+        ------
+        ThanoSQLValueError
+            - If values is empty.
+            - If page_size is not between 1 and 100 (inclusive).
+            - If max_results is not between 0 and 100 (inclusive).
+            - If the formats of template and values don't match.
+
+        """
+        if len(values) == 0:
+            raise ThanoSQLValueError("Values cannot be empty")
+
+        if page_size < 1 or page_size > 100:
+            raise ThanoSQLValueError(
+                "Page size can only range from 1 to 100 (inclusive)"
+            )
```

Comment:
Since this pagination is for the request, not the response, I don't think there is a need to limit the maximum value.